src/core/time.ts

```diff
@@ -118,6 +118,9 @@ export class TimeValue {
    * @param other The time value to add to this one.
    */
   add(other: TimeValue): TimeValue {
+    if (other.isNever()) {
```

Comment:
If we agree that `t` + -:infinity: = `t`, then -:infinity: + `t` = `t` should also hold. This latter case is not handled currently.

As stated below, I think we should also ensure that `t` + :infinity: = :infinity: + `t` = :infinity:.

Reply:
Note that if we don't encode "nothing" as -:infinity:, then I think we should have `t` + -:infinity: = -:infinity: + `t` = -:infinity: also.

Reply:
This is addressed in [26d0a9f](https://github.com/lf-lang/reactor-ts/pull/186/commits/26d0a9f240c3619d9d8b899d163697047116bd7c). `t` + -:infinity: = -:infinity: + `t` = -:infinity:, `t` + ♾️ = ♾️ + `t` = ♾️, and -:infinity: + ♾️ = 0 now. Additionally, 'no delay' is represented as `undefined` and overflow is handled either.